devito/ops/operator.py

```diff
@@ -29,6 +30,13 @@ def __init__(self, *args, **kwargs):
     def _specialize_iet(self, iet, **kwargs):
         warning("The OPS backend is still work-in-progress")
 
+        # If there is no iteration tree, then there is no loop to be optimized using OPS.
+        iteration_tree = retrieve_iteration_tree(iet, mode='normal')
+        if not len(iteration_tree):
+            return iet
+        time_upper_bound = iteration_tree[0].dimensions[TimeFunction._time_position]\
```

Comment:
for the next PR: assuming that `TimeFunction._time_position` corresponds to the first entry in the tree is generally wrong (though often true in practice)